front/src/applications/stdcm/utils/checkStdcmConfigErrors.ts

```diff
@@ -30,6 +30,17 @@ const checkStdcmConfigErrors = (
   const isOriginRespectDestinationSchedule =
     origin.arrivalType === ArrivalTimeTypes.RESPECT_DESTINATION_SCHEDULE;
 
+  if (
+    'uic' in origin &&
+    'uic' in destination &&
+    'ch' in origin &&
+    'ch' in destination &&
+    origin.uic === destination.uic &&
+    origin.ch === destination.ch
+  ) {
```

Comment:
We don't use `ch` in the code (too much sncf) but `secondary_code` (editoast type).
Both `uic` and `secondary_code` are defined if `location` is defined in `StdcmPathStep` type.
As we checked l.13 that we don't have any path step without location, we can assert that we will have one at this point.
```suggestion
 if (
    origin.location!.uic === destination.location!.uic &&
    origin.location!.secondary_code === destination.location!.secondary_code
  ) {
```

Reply:
Fixed in https://github.com/OpenRailAssociation/osrd/pull/9931/commits/838a3eacbcd5845444d92857f34f2a57dfa23998.

I also put the origin / destination check before the pathfinding response check as a bit of a hotfix, otherwise it would return the standard error when pathfinding fails instead of the new error that is more specific. It might be useful to give to `checkStdcmConfigErrors` more than just a boolean in case of pathfinding failure ? If it also received the error type returned by Editoast, we could directly return the  `zero_length_path` error instead of doing the current check on the front-end side.

Reply:
Yep, you're totally right, it should be done [here](https://github.com/OpenRailAssociation/osrd/issues/9707) :)